lib/applications.js

```diff
@@ -85,41 +85,47 @@ class Application {
 //   applications - array of JSTP applications
 //
 const createAppsIndex = (applications) => {
-  const latestApps = new Map();
   const index = new Map();
 
+  const appIndex = new Map();
+
   applications.forEach((application) => {
-    let appVersions = index.get(application.name);
-    if (!appVersions) {
-      appVersions = new Map();
-      index.set(application.name, appVersions);
+    let versions = appIndex.get(application.name);
+    if (!versions) {
+      versions = [];
+      appIndex.set(application.name, versions);
+    }
+    versions.push(application);
+  });
+
+  appIndex.forEach((instances, name) => {
+    const versions = new Map();
+    index.set(name, versions);
+    if (instances.length === 1) {
+      versions.set('1.0.0', instances[0]);
```

Comment:
You should check if app has version and set 1.0.0 only if it doesn't. You discard app version here.